lib/r7b-impl/inexact.sls

```diff
@@ -8,4 +8,22 @@ infinite? log
 nan? sin
 sqrt tan
  )
-         (import (rnrs)))
+         (import (rename (rnrs) (finite? r6rs:finite?) (infinite? r6rs:infinite?) (nan? r6rs:nan?))
+                 (only (mosh) format))
+
+(define (finite? z)
+  (unless (number? z)
+    (assertion-violation 'finite? (format #f "~e required, but got ~e" 'number z) `(number ,z)))
```

Comment:
(Note for me: Perhaps I should better to create R6RS ←→ R7RS exception mapping in Yuni. R7RS has `file-error?` and `readl-error?` so the mapping should be consistent throughout compatibility layer.)

Reply:
@okuoku A semi-official 6<->7 exception mapping would be very useful, e.g. as a SRFI. It could even be incorporated into R7RS-large.